### PR TITLE
✨ Cache list service request in catalog

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

This PR follows #2869 and the idea is to provide caching of the request listing services from the catalog using for now a simple cache mechanism through [aiocache](https://github.com/aio-libs/aiocache) library.
This library could allow caching through Redis if need be. for now it caches in memory for a 30 seconds time.

The requests per seconds doubles using that mechanism, during the 30seconds the cache lives.

Without cache:
```
------------------------------------------------------------------------------------------------ benchmark: 2 tests ------------------------------------------------------------------------------------------------
  Name (time in ms)                             Min                   Max                  Mean             StdDev                Median                 IQR            Outliers     OPS            Rounds  Iterations
  --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  test_list_services_without_details       620.0175 (1.0)        748.3653 (1.0)        666.8922 (1.0)      63.0435 (1.05)       623.8694 (1.0)      108.3981 (1.07)          1;0  1.4995 (1.0)           5           1
  test_list_services_with_details        1,270.5686 (2.05)     1,403.4215 (1.88)     1,348.9908 (2.02)     60.2470 (1.0)      1,384.6504 (2.22)     100.8497 (1.0)           1;0  0.7413 (0.49)          5           1
  --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```
With cache:
```
-------------------------------------------------------------------------------------------- benchmark: 2 tests -------------------------------------------------------------------------------------------
  Name (time in ms)                           Min                 Max                Mean             StdDev              Median                IQR            Outliers     OPS            Rounds  Iterations
  -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  test_list_services_without_details     676.1354 (1.0)      787.9862 (1.0)      716.5263 (1.0)      42.6909 (1.24)     705.7252 (1.0)      44.0494 (1.50)          1;0  1.3956 (1.0)           5           1
  test_list_services_with_details        756.4908 (1.12)     843.0671 (1.07)     782.8207 (1.09)     34.4919 (1.0)      771.2704 (1.09)     29.3169 (1.0)           1;1  1.2774 (0.92)          5           1
  -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

File: packages/settings-library/src/settings_library/postgres.py
Comment (pcrespov) [inline packages/settings-library/src/settings_library/postgres.py:67]: THOUGHT: Since we are going to deprecate aiopg this means that this will be the final ``dsn``. 
For that reason, I would directly implement it properly as

```python
    @cached_property
    def dsn_with_async_sqlalchemy(self) -> str:
        dsn: str = PostgresDsn.build(
            scheme="postgresql+asycnpg",
            user=self.POSTGRES_USER,
            password=self.POSTGRES_PASSWORD.get_secret_value(),
            host=self.POSTGRES_HOST,
            port=f"{self.POSTGRES_PORT}",
            path=f"/{self.POSTGRES_DB}",
        )
        return dsn
```
You can even add a deprecation warning if using ``dsn`` ...

The ``replace`` in addition is dangerous ... for instance, what if any of the attributes, i.e. user or password ,etc is ``postgresql``??

File: services/catalog/src/simcore_service_catalog/api/routes/services.py
Comment (pcrespov) [inline services/catalog/src/simcore_service_catalog/api/routes/services.py:73]: wild :-D ...

File: services/catalog/src/simcore_service_catalog/api/routes/services.py
Comment (pcrespov) [inline services/catalog/src/simcore_service_catalog/api/routes/services.py:139]: MINOR: the name should not incorporate ``cached_`` since this is a decorated property (can have it or not)
Q: here you do not worry about the key-name? does this mean that different requests to this entry will reuse this cache? it so, does the access of different request might play a role here?

**Review by pcrespov — COMMENTED** [review]
I'll wait until you merge previous PR to finish the review. For the moment, here are some inputs

File: services/catalog/src/simcore_service_catalog/api/routes/services.py
Comment (sanderegg) [reply services/catalog/src/simcore_service_catalog/api/routes/services.py:82]: they don't have the same TTL
actually the call to the director-v0 is always returning the same info unless the registry changed.

File: services/catalog/src/simcore_service_catalog/api/routes/services.py
Comment (sanderegg) [reply services/catalog/src/simcore_service_catalog/api/routes/services.py]: aiocache default key uses module name, method name and so on. here there is no key to worry about. and all users use the same one

**Review by pcrespov — APPROVED** [review]

File: services/catalog/src/simcore_service_catalog/api/routes/services.py
Comment (GitHK) [inline services/catalog/src/simcore_service_catalog/api/routes/services.py:42]: Question, why not raise this even more? Since this just returns the list of existing services? You could cache for several minutes. It's not like services get released so fast.

Also an endpoint can be added for internal platform usage to `bust` the cache if needed.

**Review by GitHK — APPROVED** [review]
Nice.

File: services/catalog/src/simcore_service_catalog/api/routes/services.py
Comment (sanderegg) [reply services/catalog/src/simcore_service_catalog/api/routes/services.py:42]: it's not only new services in the registry, but you can also change the access rights, the names, thumbnail and this kind of stuff. therefore I would not extend this too much.

**Pull request merged**
